src/ert/_c_wrappers/enkf/enkf_obs.py

```diff
@@ -26,18 +43,61 @@
 
 DEFAULT_TIME_DELTA = timedelta(seconds=30)
 
+USE_NEW_PARSER_BY_DEFAULT = True
 
-class ObservationConfigError(ConfigValidationError):
-    @classmethod
-    def get_value_error_message(cls, info: ErrorInfo) -> str:
-        return (
-            (
-                f"Parsing observations config file `{info.filename}` "
-                f"resulted in the following errors: {info.message}"
-            )
-            if info.filename is not None
-            else info.message
-        )
+if (
+    "USE_OLD_ERT_OBS_PARSER" in os.environ
+    and os.environ["USE_OLD_ERT_OBS_PARSER"] == "YES"
+):
+    USE_NEW_PARSER_BY_DEFAULT = False
+
+
+class _DictConfInstance:
+    def __init__(self, name: str, values: Mapping[str, Any]):
+        self.name = name
+        self.values = values
+
+    def __contains__(self, name: str) -> bool:
+        return name in self.values
+
+    def __getitem__(self, name: str) -> Any:
+        return self.values[name]
+
+
+class _HistoryConfInstance(_DictConfInstance):
+    def __init__(self, name: str, values: HistoryValues):
+        self.name = name
+        self.values = values
```

Comment:
Why not just `super().__init(name, values)` ? in place of the explicit member assignment?

Reply:
I updated it to use the inherited constructor.